Sources/NIO/EventLoopFuture.swift

```diff
@@ -461,8 +461,10 @@ extension EventLoopFuture {
     ///     - callback: Function that will receive the value of this `EventLoopFuture` and return
     ///         a new value lifted into a new `EventLoopFuture`.
     /// - returns: A future that will receive the eventual value.
-    public func thenThrowing<U>(file: StaticString = #file, line: UInt = #line, _ callback: @escaping (T) throws -> U) -> EventLoopFuture<U> {
-        return self.then(file: file, line: line) { (value: T) -> EventLoopFuture<U> in
+    public func thenThrowing<U>(file: StaticString = #file,
```

Comment:
Why is this still `thenThrowing`? Also, should we rename the generic type parameter `U` to something else, while we're giving generic type parameters better names?

Reply:
@Lukasa so `thenThrowing` is badly named and there's nothing in `Result` that does the same thing. I wanted to rename `thenThrowing` in a separate PR so we can bike shed the name with the community. I will open as soon as this is merged, is that okaY?

Reply:
I thought I replied to this but apparently I haven't: I want to do this in a separate PR because there's no prior art for `thenThrowing` and the name is bad. So I want a PR to just do this and bike shed the name with the community (after this has been merged).